pkg/registry/registry.go

```diff
@@ -38,14 +38,15 @@ const DefaultServerImage = "quay.io/openshift/origin-operator-registry"
 var ServerImage string
 
 type registry struct {
-	log      *logrus.Entry
-	client   wrapper.Client
-	reader   datastore.Reader
-	source   string
-	packages string
-	key      types.NamespacedName
-	image    string
-	address  string
+	log          *logrus.Entry
+	client       wrapper.Client
+	reader       datastore.Reader
+	source       string
+	packages     string
+	key          types.NamespacedName
+	isOpsrcOwner bool
```

Comment:
This variable name is confusing. What is its purpose, and why is it tied to an instance of the `registry` type? Is it because we happen to instantiate the `registry` separately for the `csc` controller and the `opsrc` controller? If so, it seems like that is a coincidental concern and we should be more respectful of the purpose of these objects and their composition.

Reply:
How would you propagate this information to the builders though?

Reply:
Fact of the matter is that the registry needs to know which label to populate the resources with. But I do agree the variable name is confusing? `forOpSrc` might be better but I don't think that will satisfy @kevinrizza.

>If so, it seems like that is a coincidental concern and we should be more respectful of the purpose of these objects and their composition.

What objects are you talking about here? The `registry` or the child objects?

Reply:
I mean the `registry` object. I think the fact that we are setting a field that defines if "that" `registry` object is tagged to create resources from `opsrc`es or `csc`s is inherently bad. That sort of parameter is something that should be passed along as a function parameter, not tied to the object. The fact that those `registry` objects are initialized by each reconciler is an implementation detail, and one that could change.